src/app/core/market/order-status-notifier/order-status-notifier.service.ts

```diff
@@ -35,56 +36,64 @@ export class OrderStatusNotifierService implements OnDestroy {
     if (orders.orders.length === 0) {
       return;
     }
-
-    if (!this.oldOrders || this.oldOrders.length === 0) {
-      this.oldOrders = orders.bid.orders;
-    } else {
-      this.checkOrders(orders.bid.orders);
-      this.oldOrders = orders.bid.orders;
+    if (this.oldOrders && this.oldOrders.length) {
+      this.checkOrders(orders.bid.orders.reverse());
     }
+    this.oldOrders = orders.bid.orders.reverse();
   }
 
   private notifyNewStatus(newOrder: any) {
     this.listingService.get(newOrder.listingItemId).subscribe(response => {
      newOrder.listing = response;
 
      const message = this.getMessage(newOrder.listing.title, newOrder.messages.action_button);
-     this._notification.sendNotification(message);
+       this._notification.sendNotification(message);
     });
   }
 
   checkOrders(newOrders: any) {
-    this.oldOrders.forEach(order => {
-      newOrders.forEach(newOrder => {
-        if (this.compareOrder(order, newOrder)) {
-          this.notifyNewStatus(newOrder);
-        }
+    let compOrders = [];
+    // For New Orders
+    if (this.oldOrders.length !== newOrders.length) {
+      compOrders = this.checkForNewOrders(newOrders);
+    }
+    // OldOrders that are changed
+    this.oldOrders.forEach(oldOrder => {
+      compOrders.push(newOrders.find(newOrder => this.compareOrder(oldOrder, newOrder)))
```

Comment:
Let me think about this one for a while.

I would approach it in a different way. Take the newOrders, then remove all the oldOrders by your comparison function. 

compOrders = newOrders - oldOrders (using compareOrder).